### ISSUE TITLE
Add a standalone example showing how to create new analysis tools

### ISSUE THREAD

**Issue by prasun3** [DynamoRIO/dynamorio#5879]
**Add a pointer to any prior users list discussion.**
https://groups.google.com/g/dynamorio-users/c/He1h6RCk-W0

**Is your feature request related to a problem? Please describe.**
We can have a small example that can be helpful in getting started when writing new analysis tools.

**Describe the solution you'd like**
A simple analyzer and the corresponding cmake setup.

Do you have any implementation in mind for this feature?

**Describe alternatives you've considered**

**Additional context**

**Comment by derekbruening** [comment]
So we already have a test of building an analyzer and an existing tool separately from our project (via ctest --build-and-test): the code is here https://github.com/DynamoRIO/dynamorio/blob/master/clients/drcachesim/tests/analyzer_separate.cpp and https://github.com/DynamoRIO/dynamorio/blob/master/clients/drcachesim/tests/CMakeLists.txt.  In the users list I was suggesting augmenting the documentation regarding the include path; pointing at a complete example does sound good too.  Probably it should build off what we already have (linked above), and be exported it in the release package.

**Comment by prasun3** [comment]
I didn't realize a test already existed. It makes sense to use it. How do I run it? I see it gets built in `clients/drcachesim/tests/build_and_test` but I don't see anything in `ctest -N` output that might correspond to it.

The page doesn't explicitly mention which header file to include, so I was looking at the existing tools. Looking back at the page, it does mention "drmemtrace/ headers"

> A new CMake function is defined in the DynamoRIO package which sets the include directory for using the drmemtrace/ headers

**Comment by derekbruening** [comment]
`drmemtrace_proj`:
```
$ git grep -B 1 build-and-test | grep memtrace
suite/tests/CMakeLists.txt-  add_test(drmemtrace_proj ${CMAKE_CTEST_COMMAND}
suite/tests/CMakeLists.txt:    --build-and-test "${drmemtrace_src_dir}" "${drmemtrace_build_dir}/build_and_test"
```

**Comment by prasun3** [comment]
Thanks.

If I try including raw2trace.h from this example (for module_mapper_t), I get this error `fatal error: instru.h: No such file or directory`

So I added this line to clients/drcachesim/CMakeLists.txt
```
install_client_nonDR_header(drmemtrace tracer/instru.h)
```

Then I got an error due to dr_allocator.h not being available. I added that to CMakeLists, then I got an error due to missing archive_ostream.h

What's the solution here?

**Comment by derekbruening** [comment]
Hmm, I wonder if adding includes to raw2trace.h over time without a regression test for an external project including that header using cmake has broken that usage path.  But first a higher level question: is this for legacy traces?  New traces have encodings built in and don't need module_mapper_t.

**Comment by prasun3** [comment]
Yes, it is mostly for legacy traces, but I also have a little utility to list the bins/libs from modules.log in order to archive them in case they are needed for later analysis.
